spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/OptimMethod.scala

```diff
@@ -63,6 +63,17 @@ trait OptimMethod[@specialized(Float, Double) T] extends Serializable {
    * @param config a table contains the hyper parameter.
    */
   def getHyperParameter(config: Table): String = ""
+
+  protected def getRegularizer(config: Table, weights: Tensor[T], weightsSign: Tensor[T]) = {
+    val rType = config.get[String]("RegularizationType").getOrElse("L2")
```

Comment:
That's actually a good idea. Treat "weightDecay" as an alias of "l2WeightDecay" to avoid the API break?